File: python/Penyimpanan.py
from Komponen import Komponen

class Penyimpanan(Komponen):
    def __init__(self, tipe_penyimpanan="", kapasitas=0, nama="", merk="", tahun_rilis=0, harga=0):
        super().__init__(nama, merk, tahun_rilis, harga)
        self.tipe_penyimpanan = tipe_penyimpanan
        self.kapasitas = kapasitas
    
    def setTipePenyimpanan(self, tipe_penyimpanan):
        self.tipe_penyimpanan = tipe_penyimpanan
    
    def setKapasitas(self, kapasitas):
        self.kapasitas = kapasitas
    
    def getTipePenyimpanan(self):
        return self.tipe_penyimpanan
    
    def getKapasitas(self):
        return self.kapasitas

File: cpp/Penyimpanan.cpp
#pragma once
#include "Komponen.cpp"
#include <iostream>
#include <string>
using namespace std;

class Penyimpanan : public Komponen
{
private:
    string tipe_penyimpanan;
    int kapasitas;

public:
    Penyimpanan() {}

    Penyimpanan(string tipe_penyimpanan, int kapasitas, string nama, string merk, int tahun_rilis, int harga) : Komponen(nama, merk, tahun_rilis, harga)
    {
        this->tipe_penyimpanan = tipe_penyimpanan;
        this->kapasitas = kapasitas;
    }

    void setTipePenyimpanan(string tipe_penyimpanan)
    {
        this->tipe_penyimpanan = tipe_penyimpanan;
    }

    void setKapasitas(int kapasitas)
    {
        this->kapasitas = kapasitas;
    }

    string getTipePenyimpanan()
    {
        return tipe_penyimpanan;
    }

    int getKapasitas()
    {
        return kapasitas;
    }

    ~Penyimpanan() {}
};